### DISCUSSION TITLE
0.9.3 Release Candidate

### DISCUSSION THREAD

**Discussion by mbastian** [gephi/gephi#2440] [Announcements]
We're close from releasing the 0.9.3 version of Gephi. You can see the list of changes below. Before making the release final, we want to identify any new issues, bugs or regressions that might have been introduced. We therefore ask users already familiar with Gephi to give it a try.

## How to try it out?

Use the following links depending on your platform:

[gephi-0.9.3-SNAPSHOT-windows.exe](https://oss.sonatype.org/service/local/artifact/maven/content?r=snapshots&g=org.gephi&a=gephi&v=0.9.3-SNAPSHOT&c=windows&p=exe) (Windows)

[gephi-0.9.3-SNAPSHOT-macos.dmg](https://oss.sonatype.org/service/local/artifact/maven/content?r=snapshots&g=org.gephi&a=gephi&v=0.9.3-SNAPSHOT&c=macos&p=dmg) (Mac OS X)

[gephi-0.9.3-SNAPSHOT-linux.tar.gz](https://oss.sonatype.org/service/local/artifact/maven/content?r=snapshots&g=org.gephi&a=gephi&v=0.9.3-SNAPSHOT&c=linux&p=tar.gz) (Linux)

**Note that plugins aren't supported in pre-releases. They will be supported as usual in the final release.**

### New features
- Added _Statistical Inference_ community detection algorithm, a more advanced alternative to Modularity [#2431](https://github.com/gephi/gephi/pull/2431)
- Windows and Linux versions now also embed JRE so that it isn't needed anymore to install Java separately [#2336](https://github.com/gephi/gephi/issues/2336)
- [FlatLaf](https://www.formdev.com/flatlaf/) look and feel is now used by default on all platforms [#2415](https://github.com/gephi/gephi/issues/2415)
- Appearance attributes like colors used in partition are now saved in the project [#1861](https://github.com/gephi/gephi/issues/1861)
- Last Export settings are now saved as Preferences so they persist between session [#1790](https://github.com/gephi/gephi/issues/1790)

### Bugfixes
- Improve UI to better suit HighDPI displays [#2332](https://github.com/gephi/gephi/issues/2332)
- Graph Toolbar items overlap on smaller screen sizes [#765](https://github.com/gephi/gephi/issues/765)
- Some layouts act on settled nodes [#2385](https://github.com/gephi/gephi/issues/2385)
- Ignoring GraphML desc standard tags [#2030](https://github.com/gephi/gephi/issues/2030)
- java.util.MissingResourceException gephi version 0.9.2 [#2063](https://github.com/gephi/gephi/issues/2063)
- NullpointerException on filtering datalab columns that have empty attribute values [#2015](https://github.com/gephi/gephi/issues/2015)
- Leaving "Create missing nodes" unchecked does not have the desired effect [#1878](https://github.com/gephi/gephi/issues/1878)
- 'Interval' not copied to new workspace [#1988](https://github.com/gephi/gephi/issues/1988)
- CSV Edges file not importing because Gephi does not detect Source and Target columns [#2178](https://github.com/gephi/gephi/issues/2178)
- ArrayIndexOutOfBoundsException on when deleting nodes [#1623](https://github.com/gephi/gephi/issues/1623)
- No mode indicator in Appearance panel in Gephic Mac 0.9.2 [#2098](https://github.com/gephi/gephi/issues/2098)
- Neighbors Network filter freezes with max depth [#2099](https://github.com/gephi/gephi/issues/2099)
- Unable to import a large file [#1841](https://github.com/gephi/gephi/issues/1841)
- Update crash reporter and make it GDPR compliant [#2340](https://github.com/gephi/gephi/issues/2340)
- ClassCastException: org.gephi.graph.api.types.TimestampSet cannot be cast to org.gephi.graph.api.types.TimeMap [#2297](https://github.com/gephi/gephi/issues/2297)
- Support INF, -INF in GEXF for double-type ±∞ values [#2158](https://github.com/gephi/gephi/issues/2158)
- Node betweenness is negative for big graphs when normalized [#2139](https://github.com/gephi/gephi/issues/2139)
- generatePalette is wrong [#2112](https://github.com/gephi/gephi/issues/2112)
- Loading gexf file with 0-weighted edges crashes import [#1945](https://github.com/gephi/gephi/issues/1945)
- Crash when exporting VNA graph file [#1909](https://github.com/gephi/gephi/issues/1909)
- Import CSV error edges [#1848](https://github.com/gephi/gephi/issues/1848)
- Import of graphml still confuses d3 and label fields [#1840](https://github.com/gephi/gephi/issues/1840)
- Add support of Byte Order Mark to CSV parser [#1815](https://github.com/gephi/gephi/issues/1840)
- CSV files are no longer imported correctly when double quotes inside strings are delimited with backslashes [#1812](https://github.com/gephi/gephi/issues/1812)
- NullPointerException on EdgeTypeFilter [#1811](https://github.com/gephi/gephi/issues/1811)
- GephiFormatException can cause ArrayIndexOutOfBoundsException: 0 [#1810](https://github.com/gephi/gephi/issues/1810)
- Exception with no-merge strategy in some cases. Incompatible edge should not be created [#1802](https://github.com/gephi/gephi/issues/1802)
- NullPointerException: The fileObject parameter cannot be null [#1789](https://github.com/gephi/gephi/issues/1789)
- GephiFormatException: Gephi failed saving the project.[ #1788](https://github.com/gephi/gephi/issues/1788)
- Edge labels not retained on graphml export [#1516](https://github.com/gephi/gephi/issues/1516)
- Unhide toggle global/local ranking button [#2146](https://github.com/gephi/gephi/issues/2146)
- Unwanted and inconsistent behavior for "node size" [#2224](https://github.com/gephi/gephi/issues/2224)
- Infinite repulsion in ForceAtlas with negative weights [#1714](https://github.com/gephi/gephi/issues/1714)
- Citation for PageRank algorithm [#1944](https://github.com/gephi/gephi/issues/1944)
- Dynamic attributes for dynamic network not properly merged with multiple files [#1885](https://github.com/gephi/gephi/issues/1885)
- Add 'postProcess()' method to Preview's Renderer SPI [#1490](https://github.com/gephi/gephi/issues/1490)
- Changes in labels not rendered in preview mode [#1877](https://github.com/gephi/gephi/issues/1877)
- Sum edges merge strategy gives surprising results [#2038](https://github.com/gephi/gephi/issues/2038)
- Use timezone in GEXF import, export and Create time interval in Data Laboratory [#1864](https://github.com/gephi/gephi/issues/1864)
- Overview viewport too small [#1518](https://github.com/gephi/gephi/issues/1518)
- Wrong edge coloring of selected nodes for directed graphs [#2259](https://github.com/gephi/gephi/issues/2259)
- Modularity index groups are allways 0-index based. [#1977](https://github.com/gephi/gephi/issues/1977)

## How to give feedback?

Either on this discussion or by creating a [new issue](https://github.com/gephi/gephi/issues/new/choose).

<hr /><em>This discussion was created from the release <a href='https://github.com/gephi/gephi/releases/tag/'>0.9.3</a>.</em>

**Comment by rzo1** [comment]
Tested with the bundled JRE (11) on Ubuntu 20.04 LTS. Loading a > 120k N (500k E) graph worked just fine. Played around with some mini graphs and couldn't find any big issues.

Any chance to support Java 17 without warnings / errors? 

```
März 16, 2022 2:27:41 PM org.netbeans.ProxyURLStreamHandlerFactory register
SCHWERWIEGEND: No way to find original stream handler for jar protocol
java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.net.URLStreamHandler java.net.URL.handler accessible: module java.base does not "opens java.net" to unnamed module @53c4a559
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at org.netbeans.ProxyURLStreamHandlerFactory.register(ProxyURLStreamHandlerFactory.java:59)
	at org.netbeans.JarClassLoader.<clinit>(JarClassLoader.java:117)
	at org.netbeans.MainImpl.execute(MainImpl.java:153)
	at org.netbeans.MainImpl.main(MainImpl.java:60)
	at org.netbeans.Main.main(Main.java:58)

WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.netbeans.TopSecurityManager (file:/home/zowallar/Downloads/gephi-0.9.3-20220315.194445-318-linux/gephi-0.9.3-SNAPSHOT/platform/lib/boot.jar)
WARNING: Please consider reporting this to the maintainers of org.netbeans.TopSecurityManager
WARNING: System::setSecurityManager will be removed in a future release
```

**Reply by mbastian** [reply]
Nice to hear! Thanks for the report. We indeed have a couple of Warnings that we need to take care but some are from Netbeans Plartform and I would need to check whether they are working on it. No immediate action though, we will ship with those warnings.

During development, I noticed a couple of issues with Java 17 so we preferred to stick to 11 for now. We shall revisit that again for next release 0.9.4 and see if Java 17 can be used as default.

**Reply by rzo1** [reply]
Fine. I am very happy that there will be a new release! Thanks for your (and all of the other contributors) efforts and time .

**Comment by wetneb** [comment]
Any reason why you are incrementing the patch release number and not the minor / major ones? Looking at the list of changes, there are some important new features / improvements, so from my end I would expect 0.10.0 more than 0.9.3 (but maybe that's too late now that this RC is out).

**Reply by mbastian** [reply]
Yes you're right. We are following our legacy at this point. I think once we have a 1.0 we can follow the more standard versioning. It might be too confusing to switch now as we've always marked major released as increment in the "minor": 0.8, 0.9 etc.

**Reply by mbastian** [reply]
For the record, the discussion should continue here: https://github.com/gephi/gephi/issues/2444

**Comment by NuLL3rr0r** [comment]
Thank you very much for this release! I've been waiting for this release for years.

**Comment by jacomyma** [comment]
I found an issue with CSV import. With the same file, it works well in 0.9.2:
![image](https://user-images.githubusercontent.com/158525/159116204-cf336c9e-2fcc-4dbe-b8d0-05fd0410e6de.png)
but the file is not parsed correctly in 0.9.3:
![image](https://user-images.githubusercontent.com/158525/159116279-b962851b-ccb1-4af7-bb62-64087c893751.png)
It does not generate an error, but the CSV is not interpreted as it should.

**Reply by mbastian** [reply]
Thanks @jacomyma, please share the CSV with me here or over Slack and I'll look into it!

**Reply by jacomyma** [reply]
So for the record:

* The CSV has an issue (last column missing a header)
* CSV files without this error are parsed correctly
* Although the 0.9.3 behavior is different from 0.9.2, it does not mean that it is not desirable. Indeed, having a malformed CSV trigger an error at parsing is nothing wrong by itself.

So in short, arguably, there is nothing to see here.

**Comment by Haaroon** [comment]
A release after 5 years?? Very surprising, very refreshing!

**Reply by acralkmim** [reply]
How to use the streaming twitter pra other importer API v2.

**Comment by ngmaclaren** [comment]
Testing on Manjaro 21.2.5. My normal tasks seem quicker with this version and everything works as expected. Outstanding!

**Reply by ngmaclaren** [reply]
I have some warnings as well in case they're informative. These are startup warnings from running `zsh ./gephi` from inside `[SNAPSHOT]/bin/`:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.netbeans.ProxyURLStreamHandlerFactory (file:/home/neil/gephi-0.9.3-20220327.193704-345-linux/gephi-0.9.3-SNAPSHOT/platform/lib/boot.jar) to field java.net.URL.handler
WARNING: Please consider reporting this to the maintainers of org.netbeans.ProxyURLStreamHandlerFactory
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Reply by paulgirard** [reply]
I've got the same warnings but it works perfectly for the little test made on ubuntu 20.04 lunix mint flavor. 
Congrats !
Next week I am participating a workshop where I might ind Mac Os and Windows users. I might ask them to test this release.

**Reply by mbastian** [reply]
Thank you. These warnings cause no harm, feel free to ignore them. We need to look deeper into it but not super high priority.

**Comment by sarepal** [comment]
Tested the pre-release with a small network and found that a bug I was experiencing in 0.9.2 (erroneous edge in the graph layout) has been resolved. Eager for the full release as I need GeoLayout for this work. When might the full release come out?

**Reply by mbastian** [reply]
Cool! If all goes well, early next week. Plugins shortly after that.

**Comment by acralkmim** [comment]
How to streaming twitter API 2 whith the New version?

**Reply by mbastian** [reply]
Hi @acralkmim, I'll let @totetmatt address questions related to the Twitter plug-in

**Comment by guillaumebuffet** [comment]
Hi there. A can't wait... but have to as, once Gephi 0.9.3 installed, it's impossible to access existing files / import files from my computer (Mac OS 12.2.1). 
Seems that Gephi hasn't he permission to access my files. Works perfectly with previous version. 

Uninstalled and re-installed. Still the same.

I've tried with 0.9.2. When I re-install, I've a permission windows to access my files. Not with 0.9.3. 

An idea ?

Thanks for your help.

Guillaume
<img width="720" alt="image" src="https://user-images.githubusercontent.com/58593069/162734349-11a3643e-c970-4091-ad78-65f9f2dacc7b.png">

**Reply by eduramiba** [reply]
@guillaumebuffet See #2467

**Reply by guillaumebuffet** [reply]
Thanks !

**Reply by guillaumebuffet** [reply]
Sauvé ! Fantastic ! Much - much - faster than the previous version.
Guillaume
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/58593069/162756170-dfc92856-1294-45b1-897d-d0dd74e862a1.png">

**Comment by Splines** [comment]
Thank you so much for this update and all your efforts you put into this amazing program, the new look & feel is great and I also appreciate that Java comes bundled now.
